platform/security/sts/security-sts-server/src/main/java/ddf/security/sts/AuthNStatementProvider.java

```diff
@@ -46,7 +46,13 @@ public AuthNStatementProvider() {
   @Override
   public AuthenticationStatementBean getStatement(TokenProviderParameters providerParameters) {
     AuthenticationStatementBean authBean = new AuthenticationStatementBean();
-    authBean.setSessionIndex(Integer.toString(secureRandom.nextInt()));
+    // "Use small positive integers (or reoccurring constants in a list) for the SessionIndex"
```

Comment:
❓ I couldn't quite figure out by reading the spec, but I thought session indices need to be unique for a given Principal? If so, do we have enough resolution here to avoid collisions?

Reply:
We could use a single atomically incremented integer that loops back around to 1 If we're worried about the space being too small for collisions with a random. Although "small" seems ambiguous with out any relative comparison - it could be safest to just  broaden the range a bit.

Reply:
The spec says not to use numbers sequential numbers because it conveys some advantage to attackers. I wasn't so clear on what that advantage could be.

Session indexes do not have to globally unique. Once they just have to be unique for a given Principal. Typically, a Principle only has one SessionIndex. The day may come where they gave 2 or 3, but I don't see a collision as likely.

Reply:
There is `LogoutRequestServiceTest.testGetLogoutRequestNoSessionIndex`. It isn't working the way I thought it was, so I'm pushing a fix for it soon.

Reply:
I guess that's the problem with random numbers. It is _unlikely_ that they will collide but not _impossible_. My worry here is that _when_ it happens, it will be very difficult to know what's going on and almost impossible to reproduce. Increasing the number space would help reduce the collision likelihood and might be good enough, but it all depends on the random generator's number distribution over time and such, which can be tricky.

Reply:
One thing we could consider here, since we know that we're not expecting hundreds of sessions per Principal, would be to populate a `Set` with _N_ random numbers to guarantee uniqueness, copy those values to a list, use `Collections.shuffle()` to randomize the order, and use Guava's `Iterables.cycle()` to get unique, pseudo-random session indices.

Reply:
I'm happy to increase the space. I'll double the number of bytes from 4 to 8. Does that sound OK?

Reply:
I'm ok with postponing this change and doing that work in a separate ticket, but I'd recommend creating the ticket now and get it done as part of the SAML Compliance work as soon as we can.